code/parse/parse_manager.h

```diff
@@ -0,0 +1,60 @@
+#include "globalincs/systemvars.h"
+#include "globalincs/pstypes.h"
+
+enum class Parse_Input_Types : std::uint8_t
+{
+	USAGE_NOTE,
+	STRING,
+	XSTR,
+	INT,
+	INT_LIST,
+	LONG,
+	FLOAT,
+	FLOAT_LIST,
+	DOUBLE,
+	VEC2,
+	VEC3,
+	BOOL,
+	FLAGS,
+	MARKER,
+	MODEL_FILENAME,
+	TEXTURE_FILENAME,
+	SOUND_FILENAME
+};
+
+class parse_item {
+protected:
+	// basic description 
+	SCP_string	_table_string;
+	SCP_string	_description;
+
+	Parse_Input_Types	_type;
+	// rules for this element
+	bool		_required;
+	int			_enforced_count;		// if this is a list that has a specific number of entries, mark it with this variable
+	int			_major_version_number_implemented;
+	int			_minor_version_number_implemented;
+	int			_revision_number_implemented;
+	int			_major_version_number_deprecated;
+	int			_minor_version_number_deprecated;
+	int			_revision_number_deprecated;
+	SCP_string	_deprecation_message;
+
+public:
+	parse_item() = default;
+	// our factory functions
+	parse_item(SCP_string* table_string);
+	parse_item set_description(SCP_string* description);
+	parse_item set_type(Parse_Input_Types type);
+	parse_item set_required(bool required);
+	parse_item set_enforced_count(int enforced_count);
+	parse_item set_implementation_version(int major_implementation_version, int minor_implementation_version, int revision_implementation_number);
+	parse_item set_deprecation_version(int major_deprecation_version, int minor_deprecation_version, int revision_deprecation_number);
+	parse_item set_deprecation_message(SCP_string* deprecation_message);
```

Comment:
If this is supposed to be a factory class, then you probably also want to return by reference here (and above).

Reply:
I think this is handled yet.  Please tell me if I misunderstood.